dandiapi/api/views/upload.py

```diff
@@ -162,15 +187,20 @@ def upload_complete_view(request: Request, upload_id: str) -> HttpResponseBase:
     request_serializer.is_valid(raise_exception=True)
     parts: List[TransferredPart] = request_serializer.save()
 
-    upload = get_object_or_404(Upload, upload_id=upload_id)
+    try:
+        upload = Upload.objects.get(upload_id=upload_id)
+    except Upload.DoesNotExist:
+        upload = get_object_or_404(EmbargoedUpload, upload_id=upload_id)
+        if not request.user.has_perm('owner', upload.dandiset):
+            raise Http404()
```

Comment:
```suggestion
      upload = get_object_or_404(BaseUpload, upload_id=upload_id)
```

Would this work? Maybe you already thought of this and Django doesn't do polymorphism like that

Reply:
Actually, I guess this doesn't make sense to do because of that `if` statement checking if the user is an owner on the embargoed dandiset.